pyquil/tests/test_device.py

```diff
@@ -1,52 +1,126 @@
-                "qubit-id": 2
-            }
-        ],
-        [
-            {
-                "action-qubits": [0, 1],
-                "type": "CZ"
-            },
-            {
-                "action-qubits": [1, 2],
-                "type": "ISWAP"
-            },
-            {
-                "action-qubits": [2, 0],
-                "type": "CPHASE"
-            }
+from pyquil.device import Device, KrausModel, ISA, Qubit, Edge, NoiseModel
+
+DEVICE_FIXTURE_NAME = 'mixed_architecture_chip'
+
+
+@pytest.fixture
+def isa_dict():
+    return {
+        "id": {
+            "name": DEVICE_FIXTURE_NAME,
+            "version": "0.0",
+            "timestamp": "20180104103600"
+        },
+        "logical-hardware": [
+            [
+                {
+                    "qubit-id": 0,
+                    "type": "Xhalves"
+                },
+                {
+                    "qubit-id": 1
+                },
+                {
+                    "qubit-id": 2
+                }
+            ],
+            [
+                {
+                    "action-qubits": [0, 1],
+                    "type": "CZ"
+                },
+                {
+                    "action-qubits": [1, 2],
+                    "type": "ISWAP"
+                },
+                {
+                    "action-qubits": [2, 0],
+                    "type": "CPHASE"
+                }
+            ]
         ]
-    ]
-}
+    }
+
+
+@pytest.fixture
+def isa(isa_dict):
+    return ISA.from_dict(isa_dict)
+
+
+@pytest.fixture
+def kraus_model_I_dict():
+    return {'gate': 'I',
+            'fidelity': 1.0,
+            'kraus_ops': [[[[1.0]], [[1.0]]]],
+            'targets': (0, 1),
+            'params': (5.0,)}
+
+
+@pytest.fixture
+def kraus_model_I(kraus_model_I_dict):
+    return KrausModel.from_dict(kraus_model_I_dict)
+
+
+@pytest.fixture
+def kraus_model_RX90_dict():
+    return {'gate': 'RX',
+            'fidelity': 1.0,
+            'kraus_ops': [[[[1.0]], [[1.0]]]],
+            'targets': (0,),
+            'params': (np.pi / 2.,)}
+
+
+@pytest.fixture
+def kraus_model_RX90(kraus_model_RX90_dict):
+    return KrausModel.from_dict(kraus_model_RX90_dict)
 
 
-def test_isa():
-    isa = ISA.from_dict(ARCH_QPU)
+@pytest.fixture
+def noise_model_dict():
+    return {'gates': [{'gate': 'I',
+                       'params': (5.0,),
+                       'targets': (0, 1),
+                       'kraus_ops': [[[[1.0]], [[1.0]]]],
+                       'fidelity': 1.0},
+                      {'gate': 'RX',
+                       'params': (np.pi / 2.,),
+                       'targets': (0,),
+                       'kraus_ops': [[[[1.0]], [[1.0]]]],
+                       'fidelity': 1.0}],
+            'assignment_probs': {'1': [[1.0, 0.0], [0.0, 1.0]],
+                                 '0': [[1.0, 0.0], [0.0, 1.0]]},
+            'isa_name': DEVICE_FIXTURE_NAME}
+
+
+@pytest.fixture
+def noise_model(noise_model_dict):
```

Comment:
same as with the`isa` fixture, because this fixture is only used in `test_device` i think it would be clearer to just pass the dictionary fixture to that test and build the `NoiseModel` in the test with `NoiseModel.from_dict`. this also keeps us from calling code that we want to test from inside a fixture